### PR TITLE
impl(docfx): support `variablelist` elements

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #10895

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10924)

### CODE REVIEW

File: docfx/doxygen2markdown.cc [hunk 1]
```diff
@@ -633,6 +643,89 @@ bool AppendIfListItem(std::ostream& os, MarkdownContext const& ctx,
...
+      nested.paragraph_start = "\n";
+      nested.paragraph_indent = ctx.paragraph_indent + "- ";
+      continue;
+    }
+    UnknownChildType(__func__, child);
+  }
+  return true;
+}
+
+// A varlist entry contains a single `term` element.
```
Comment (dbolduc) [inline docfx/doxygen2markdown.cc:686]: nit: s/varlist entry/\`varlistentry\`/

File: docfx/doxygen2markdown.cc [hunk 2]
```diff
@@ -633,6 +643,89 @@ bool AppendIfListItem(std::ostream& os, MarkdownContext const& ctx,
...
+//   <xsd:complexType name="docVariableListType">
+//     <xsd:sequence>
+//       <xsd:group ref="docVariableListGroup" maxOccurs="unbounded" />
+//     </xsd:sequence>
+//   </xsd:complexType>
+//
+//   <xsd:group name="docVariableListGroup">
+//     <xsd:sequence>
+//       <xsd:element name="varlistentry" type="docVarListEntryType" />
+//       <xsd:element name="listitem" type="docListItemType" />
```
Comment (dbolduc) [inline docfx/doxygen2markdown.cc:659]: So one of these things can't have multiple `listitem`s?

If they could, then the `nested.paragraph_indent = ...` below could introduce unwanted `- `s

File: docfx/doxygen2markdown.cc [hunk 3]
```diff
@@ -414,7 +423,7 @@ bool AppendIfDocTitleCmdGroup(std::ostream& os, MarkdownContext const& ctx,
   // Unexpected: subscript, superscript, center, small, del, ins
   // Unexpected: htmlonly, manonly, rtfonly, latexonly, docbookonly
   // Unexpected: image, dot, msc, plantuml
-  // Unexpected: anchor
+  if (name == "anchor") return true;  // Generate nothing and succeed
```
Comment (devbww) [inline docfx/doxygen2markdown.cc:426]: Opt: Treat this just like the others (e.g. `AppendIfAnchor()`), even if it appends nothing?

File: docfx/doxygen2markdown.h [hunk 4]
```diff
@@ -141,6 +141,24 @@ bool AppendIfItemizedList(std::ostream& os, MarkdownContext const& ctx,
 bool AppendIfListItem(std::ostream& os, MarkdownContext const& ctx,
                       pugi::xml_node const& node);
 
+/**
+ *
+ */
```
Comment (devbww) [inline docfx/doxygen2markdown.h:146]: ???

**Review by devbww — APPROVED** [review]

File: docfx/doxygen2markdown.h (hunk 4 above)
Comment (coryan) [reply docfx/doxygen2markdown.h]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 3 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Done.

File: docfx/doxygen2markdown.h [hunk 5]
```diff
@@ -141,6 +141,24 @@ bool AppendIfItemizedList(std::ostream& os, MarkdownContext const& ctx,
...
+/// Handle `variablelist` elements.
+///
+/// Most commonly used in lists of deprecated symbols, these are pairs of
+/// terms (a linked code entity), and a sequence of text.
```
[the remark this answers is not in the document]
Comment (coryan) [reply docfx/doxygen2markdown.h]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 1 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc]: Fixed.

File: docfx/doxygen2markdown.cc (hunk 2 above)
Comment (coryan) [reply docfx/doxygen2markdown.cc:694]: I may be missing the intent of your question.  These are supposed to have exactly one `varlistentry` and exactly one `listitem`, yes.  I think the code is fine.

Comment (dbolduc) [reply docfx/doxygen2markdown.cc:694]: That answers my question.

**Pull request merged**